jwala-services/src/main/java/com/cerner/jwala/service/webserver/component/WebServerStateSetterWorker.java

```diff
@@ -113,37 +122,18 @@ public void pingWebServer(final WebServer webServer) {
     }
 
     /**
-     * Checks if the web server is in webServerToPingSet, if it's not in the set it will be added to the set.
-     * This method is thread safe.
-     * @param webServer the web server
-     * @return true if the web server is in webServerToPingSet
-     */
-    private boolean isInWebServerToPingSet(final WebServer webServer) {
```

Comment:
The problem is this method adds web server to the set which is not apparent by just looking at it. I was actually the one who wrote this and yet I got confused when I came back to do work on it. I can't really separate the "add" part since it should be under a synchronized clause.  

In addition it is an anti pattern since it does something else rather than check if the web server is in the set. I can rename the method but it seems awkward e.g. isInWebServerToPingSetAddIfItIsNot :)